lib/galaxy/tool_util/parser/factory.py

```diff
@@ -82,7 +82,10 @@ def get_input_source(content):
     consume using the tool input source interface.
     """
     if not isinstance(content, InputSource):
-        content = XmlInputSource(content)
+        if isinstance(content, dict):
```

Comment:
Can you update the the docstring for this ?

Reply:
Done locally - will push in a second.